### PR TITLE
add keypoint mnn deploy support

### PR DESCRIPTION
**Pull request by zhiboniu**
del json config & preprocess
add keypoint mnn support

平台：麒麟980
模型：tinypose256_fn
MNN:
 tinypose  min =  0.0179  max =  0.0218  avg =  0.0192(s)
LIte:
min: 0.017, max: 0.021, avg=0.019-0.020(s)

### CODE REVIEW

File: deploy/third_engine/demo_mnn_kpts/README.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+cmake -DCMAKE_TOOLCHAIN_FILE=$ANDROID_NDK/build/cmake/android.toolchain.cmake -DANDROID_ABI="arm64-v8a" -DANDROID_PLATFORM=android-21 -DANDROID_TOOLCHAIN=gcc ..
+make
+```
+
+## Run
+
+C++ inference interface is same with NCNN code, to detect images in a fold, run:
+
+``` shell
+./tinypose-mnn "1" "../imgs/test.jpg"
```
Comment (jerrywgz) [inline deploy/third_engine/demo_mnn_kpts/README.md:95]: 增加下参数说明

File: deploy/third_engine/demo_mnn_kpts/keypoint_detector.cpp [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+                 cv::Point2d(x_end, y_end),
+                 colormap[i],
+                 1);
+      }
+    }
+  }
+  return vis_img;
+}
+
+// void KeyPointDetector::Preprocess(const cv::Mat& ori_im) {
```
Comment (jerrywgz) [inline deploy/third_engine/demo_mnn_kpts/keypoint_detector.cpp]: 去除注释代码

File: deploy/third_engine/demo_mnn_kpts/keypoint_detector.hpp [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+  explicit KeyPointDetector(const std::string& model_path,
+                            int num_thread = 4,
+                            int input_height = 256,
+                            int input_width = 192,
+                            float score_threshold = 0.3,
+                            const int batch_size = 1,
+                            bool use_dark = true) {
+    printf("config path: %s",
+           model_path.substr(0, model_path.find_last_of('/') + 1).c_str());
+    // config_.load_config(model_path.substr(0, model_path.find_last_of('/') +
```
Comment (jerrywgz) [inline deploy/third_engine/demo_mnn_kpts/keypoint_detector.hpp]: 去除注释代码

File: deploy/third_engine/demo_mnn_kpts/keypoint_postprocess.cpp [hunk 4]
```diff
@@ -0,0 +1,258 @@
+//   Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserved.
```
Comment (yghstill) [inline deploy/third_engine/demo_mnn_kpts/keypoint_postprocess.cpp]: 2021-> 2022

File: deploy/third_engine/demo_mnn_kpts/README.md (hunk 1 above)
Comment (zhiboniu) [reply deploy/third_engine/demo_mnn_kpts/README.md:95]: done

File: deploy/third_engine/demo_mnn_kpts/keypoint_detector.cpp (hunk 2 above)
Comment (zhiboniu) [reply deploy/third_engine/demo_mnn_kpts/keypoint_detector.cpp]: done

File: deploy/third_engine/demo_mnn_kpts/keypoint_detector.hpp (hunk 3 above)
Comment (zhiboniu) [reply deploy/third_engine/demo_mnn_kpts/keypoint_detector.hpp]: done

File: deploy/third_engine/demo_mnn_kpts/keypoint_postprocess.cpp (hunk 4 above)
Comment (zhiboniu) [reply deploy/third_engine/demo_mnn_kpts/keypoint_postprocess.cpp]: done

**Review by yghstill — APPROVED** [review]

**Review by jerrywgz — APPROVED** [review]

**Pull request merged**